### DISCUSSION TITLE
Parallelization of souffle

### DISCUSSION THREAD

**Discussion by EvgSkv** [souffle-lang/souffle#2516] [Q&A] [answered]
Dear Souffle team,
I'm running souffle `reach` predicate on livejournal dataset in csv format written follows:

```
.decl g(a:number,b:number)
.decl reach(a:number)
.decl result(a:number)

.input g(filename="livejournal.csv",delimiter=",",headers=true)

reach(0).
reach(y) :- reach(x), g(x, y).

result(a) :- reach(a).

.output result(delimiter=",")
```

Linux `time` command shows that user and real time are the same:

```
real	0m30.796s
user	0m29.998s
sys	0m0.676s
```

This means that no parallelization is happening, right? Why would that be the case? Do I need to enable it somehow?

Thank you very much for the great language!

**Comment by cwarden** [comment, marked as answer]
Use `--jobs auto` to use all cores.

**Comment by EvgSkv** [comment]
Perfect, I see it parallelized. Thank you!

Now most time is spent reading data. Is there a better format to use for souffle data other than csv? I tried using souffle to compress csv and read that, but it didn't improve speed.

Maybe csv is optimial, that's totally fine. Please let me know! Thanks a lot!

**Reply by cwarden** [reply]
Try importing the csv into a sqlite database and using that as the input for souffle.

**Reply by EvgSkv** [reply]
I will try that! Thank you very much!
